python/vosk/aligner/multipass.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import wave
+import sys
+
+from . import metasentence
+from . import language_model
+from . import diff_align
+from . import transcription
+from .transcriber import Transcriber as transcriber
+'''The script will rework.
+Multipass realign unaligned words.
+Prepare multipass checking words sequence, when word's case ==
+not-found-in-audio preparing chunk to realign like [words before, unaligned
+words, words after], creating new personam small language and acoustic models
+for the chunk, putting back into result sequence.
+'''
+def prepare_multipass(alignment):
+    to_realign = []
+    chunks = 0
+    for i, w in enumerate(alignment):
+        if w.case == 'not-found-in-audio':
+            for j, wd in enumerate(alignment):
+                if j >= max(0, i-3) and j <= min(len(alignment), i+3):
+                    wd.realign = True
+    cur_list = []
+    start_idx = []
+    end_idx = []
+    start_counter = 0
+    end_counter = 0
+    for j, wd in enumerate(alignment):
```

Comment:
The logic of this loop is completely unclear, you need to rewrite it to make it more straightforward